pkg/controllers/node/ipam.go

```diff
@@ -458,27 +521,35 @@ func (c *ipamController) reviewIPAMState() ([]string, error) {
 				continue
 			}
 
-			// The node is ready to be removed. It exists in Calico IPAM, but
+			// The node is ready have its IPAM affinities released. It exists in Calico IPAM, but
 			// not in the Kubernetes API. Additionally, we've checked that there are no
 			// outstanding valid allocations on the node.
-			nodesToDelete = append(nodesToDelete, cnode)
+			nodesToRelease = append(nodesToRelease, cnode)
 		}
 	}
-	return nodesToDelete, nil
+	return nodesToRelease, nil
 }
 
-// checkAllocation returns true if the allocation is still in use, and false if the allocation
-// apears to be leaked.
-func (c *ipamController) checkAllocation(a *allocation, knode string) bool {
+// allocationIsValid returns true if the allocation is still in use, and false if the allocation
+// appears to be leaked.
+func (c *ipamController) allocationIsValid(a *allocation, knode string) bool {
 	ns := a.attrs[ipam.AttributeNamespace]
 	pod := a.attrs[ipam.AttributePod]
 	logc := log.WithFields(a.fields())
 
+	if ns == "" || pod == "" {
+		// Allocation is either not a pod address, or it pre-dates the use of these
+		// attributes. Assume it's a valid allocation since we can't perform our
+		// confidence checks below.
+		logc.Info("IP allocation is missing metadata, cannot confirm or deny validity. Assume valid.")
```

Comment:
Could get spammy on a cluster that's been upgraded?  Might be 1000s of IPs in that state.